crates/catalog/rest/src/catalog.rs

```diff
@@ -0,0 +1,845 @@
+        &self,
+        _namespace: &NamespaceIdent,
+        _creation: TableCreation,
+    ) -> Result<Table> {
+        Err(Error::new(
+            ErrorKind::FeatureUnsupported,
+            "Creating table not supported yet!",
+        ))
+    }
+
+    /// Load table from the catalog.
+    async fn load_table(&self, _table: &TableIdent) -> Result<Table> {
+        Err(Error::new(
+            ErrorKind::FeatureUnsupported,
+            "Creating table not supported yet!",
+        ))
+    }
+
+    /// Drop a table from the catalog.
+    async fn drop_table(&self, table: &TableIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .delete(self.config.table_endpoint(table))
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// Check if a table exists in the catalog.
+    async fn stat_table(&self, table: &TableIdent) -> Result<bool> {
+        let request = self
+            .client
+            .0
+            .head(self.config.table_endpoint(table))
+            .build()?;
+
+        self.client
+            .execute::<ErrorModel, NO_CONTENT>(request)
+            .await
+            .map(|_| true)
+    }
+
+    /// Rename a table in the catalog.
+    async fn rename_table(&self, src: &TableIdent, dest: &TableIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .post(self.config.rename_table_endpoint())
+            .json(&RenameTableRequest {
+                source: src.clone(),
+                destination: dest.clone(),
+            })
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// Update a table to the catalog.
+    async fn update_table(&self, _table: &TableIdent, _commit: TableCommit) -> Result<Table> {
+        todo!()
+    }
+
+    /// Update multiple tables to the catalog as an atomic operation.
+    async fn update_tables(&self, _tables: &[(TableIdent, TableCommit)]) -> Result<()> {
+        todo!()
+    }
+}
+
+impl RestCatalog {
+    /// Creates a rest catalog from config.
+    pub async fn new(config: RestCatalogConfig) -> Result<Self> {
+        let mut catalog = Self {
+            client: config.try_create_rest_client()?,
+            config,
+        };
+
+        catalog.update_config().await?;
+        catalog.client = catalog.config.try_create_rest_client()?;
+
+        Ok(catalog)
+    }
+
+    async fn update_config(&mut self) -> Result<()> {
+        let mut request = self.client.0.get(self.config.config_endpoint());
+
+        if let Some(warehouse_location) = &self.config.warehouse {
+            request = request.query(&[("warehouse", warehouse_location)]);
+        }
+        let mut config = self
+            .client
+            .query::<CatalogConfig, ErrorResponse, OK>(request.build()?)
+            .await?;
+
+        config.defaults.extend(self.config.props.clone());
```

Comment:
This is correct, but it looks a bit odd because you overwrite the default (which is fine)